trace/google/cloud/trace/span_context.py

```diff
@@ -0,0 +1,164 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""SpanContext encapsulates the current context within the request's trace."""
+
+from google.cloud.trace.trace import generate_trace_id
+
+import re
+import logging
+
+_TRACE_CONTEXT_HEADER_FORMAT = '([0-9a-f]{32})(\/(\d+))?(;o=(\d+))?'
+_TRACE_HEADER_KEY = 'X_CLOUD_TRACE_CONTEXT'
+_TRACE_ID_FORMAT = '[0-9a-f]{32}?'
+
+
+class SpanContext(object):
+    """SpanContext includes 3 fields: traceId, spanId, and an enabled flag 
+    which indicates whether or not the request is being traced. It contains the
+    current context to be propagate to the child spans.
+    
+    :type trace_id: str
```

Comment:
That seems like an unnecessary conversion.  We initially get the value as a string object when we parse it from inbound http headers, and the Python API for outbound grpc requests will take it from us as a string object.  Specifically we'll make grpc calls passing in a dictionary of string to string like `metadata: {'X_CLOUD_TRACE_CONTEXT': whatever.trace_id, ... other metadata ... }`.  Likewise, we'll pass it to the Python protobuf constructor as a string object when we're recording the trace to the trace service.  We don't ever have a use for it as a compact byte array.